src/dwarf_parser.cpp

```diff
@@ -256,164 +219,36 @@ void Dwarf::resolve_fields(const SizedType &type) const
   std::string type_name = type.GetName();
   if (type_name.find("struct ") == 0)
     type_name = type_name.substr(strlen("struct "));
-  auto type_die = find_type(type_name);
-  if (!type_die)
+  else if (type_name.find("class ") == 0)
+    type_name = type_name.substr(strlen("class "));
+  else if (type_name.find("union ") == 0)
+    type_name = type_name.substr(strlen("union "));
```

Comment:
This code (old and new) might be wrong given that `Foo` and `struct Foo` are different types in C: #682